### PR TITLE
Refactoring to increase test coverage, added TypeValue::isSubtypeOf method

### PR DESCRIPTION
**Pull request by lucaswoj**

### CODE REVIEW

File: src/test/java/org/whdl/frontend/syntaxtree/TestVariableIdentifier.java [hunk 1]
```diff
@@ -69,4 +75,19 @@ public void testNotEqualsNamespace() {
     );
     assertNotEquals(getInstance(), v);
   }
+  
+  @Test
+  public void toString_namespaced() {
+    assertEquals(getInstance().toString(), "whdl:foo");
```
Comment (mtrberzi) [inline src/test/java/org/whdl/frontend/syntaxtree/TestVariableIdentifier.java:81]: Try not to hardcode the separator here -- grab it from NamespaceIdentifier

Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestVariableIdentifier.java:81]: I think I can make a case for allowing this separator to be hardcoded in unit tests; if the separator changes, these tests will fail, and we will know immediately.

Obviously it is NOT ok to hardcode this separator in the "main" codebase.

Lets discuss in the standup later.

**Comment by TysonAndre** [comment]
seems fine. You could add tests for .equals(null), but those would pass anyways.

What will the type hierarchy look like?

**Comment by lucaswoj** [comment]
@TysonAndre Not sure exactly what you're referring to -- the tests of `verify`? What I'm really testing is that they don't throw any exceptions. 

We don't have much of a type hierarchy right now. Every type is a first class object that subclasses `Type`. Types include `Bit`, `Tuple`, and `Function` (and maybe `Enum` and `Integer`)

**Comment by TysonAndre** [comment]
I meant the tests of NamespaceIdentifier.equals

thanks for clarifying type hierarchy

**Comment by lucaswoj** [comment]
If this looks good, feel free to merge :smile:

**Pull request merged**